### ISSUE TITLE
Feature Request:  compression support

### ISSUE THREAD

**Issue by sweihub** [neonmoe/minreq#101]
Hi

Thanks for the cute minreq, it's really handy!

However, can you add compression support?

```rust
let records: Vec<Last> = minreq::get(url)
                .with_header("Accept-Encoding", "deflate, gzip")
                .with_timeout(5)
                .send()?
                .json()?;
```

Gives the error
```
response contained invalid utf-8 where valid utf-8 was expected
```

**Comment by ali-izadii** [comment]
Hi! I’d like to try working on this, starting with gzip support behind an optional Cargo feature. I’m not sure yet whether I can complete it, but I’d like to give it a go. Would that scope work for you? Are there any implementation preferences I should follow?
@neonmoe

**Comment by neonmoe** [comment]
Yes, that sounds good, thanks! Generally in this crate, the optional dependencies each map to a specific feature gated behind it, so I'd continue in that vein and gate the functionality behind `#[cfg(feature = "the_gzip_crate_name"]`.

Regarding API/how this functionality is triggered, since this is an Accept header based feature, I'd say we could modify/add the relevant header signaling to the server that we can handle gzipped responses, without any additional API or function to "enable" this behavior. Of course, I'm open to other approaches too if there's reasons to go another way, that's just what I'd go with without investigating this much further.

Regarding general code style, maybe some guideline: no need to split into many functions if one will do, avoid unnecessary allocations, prefer not using generics, and try to keep the code straightforward. But in the end, feel free to do whatever, I'm sure your taste already aligns with mine, given you want to contribute to this crate in particular, and it shouldn't be *that* big of a feature anyway :) If there's need to iterate, I'll give feedback on the PR.

**Comment by ali-izadii** [comment]
Hi! My PR is ready: #130. Some CI checks are failing, though, and I’m not sure why. Could you help me understand what the failures mean and how to investigate them?
@neonmoe
